pkg/dependencies/factory.go

```diff
@@ -641,6 +642,18 @@ func (f *Factory) WithHelmInsecure() *Factory {
 	return f
 }
 
+func (f *Factory) WithCertManagerClient(kubeconfig string) *Factory {
+	f.WithKubectl()
+
```

Comment:
Have a check here to see if `CertManagerClient` is already set or not so that we don't add to the buildsteps again